src/game/assets.rs

```diff
@@ -44,58 +50,64 @@ pub enum SfxAsset {
-        );
-        assets.insert(
-            SfxAsset::ButtonPress,
-            asset_server.load("audio/sfx/button_press.ogg"),
-        );
-        assets.insert(SfxAsset::Step1, asset_server.load("audio/sfx/step1.ogg"));
-        assets.insert(SfxAsset::Step2, asset_server.load("audio/sfx/step2.ogg"));
-        assets.insert(SfxAsset::Step3, asset_server.load("audio/sfx/step3.ogg"));
-        assets.insert(SfxAsset::Step4, asset_server.load("audio/sfx/step4.ogg"));
+impl AssetKey for SfxKey {
+    type Asset = AudioSource;
 
-        Self(assets)
-    }
-
-    pub fn all_loaded(&self, assets: &Assets<AudioSource>) -> bool {
-        self.0.iter().all(|(_, handle)| assets.contains(handle))
+    fn load(asset_server: &AssetServer) -> AssetMap<Self> {
+        AssetMap(HashMap::from([
+            (
+                SfxKey::ButtonHover,
+                asset_server.load("audio/sfx/button_hover.ogg"),
+            ),
+            (
+                SfxKey::ButtonPress,
+                asset_server.load("audio/sfx/button_press.ogg"),
+            ),
+            (SfxKey::Step1, asset_server.load("audio/sfx/step1.ogg")),
+            (SfxKey::Step2, asset_server.load("audio/sfx/step2.ogg")),
+            (SfxKey::Step3, asset_server.load("audio/sfx/step3.ogg")),
+            (SfxKey::Step4, asset_server.load("audio/sfx/step4.ogg")),
+        ]))
     }
 }
 
 #[derive(PartialEq, Eq, Hash, Reflect)]
-pub enum SoundtrackAsset {
+pub enum SoundtrackKey {
     Credits,
     Gameplay,
 }
 
-#[derive(Resource, Reflect, Deref, DerefMut)]
-pub struct SoundtrackAssets(HashMap<SoundtrackAsset, Handle<AudioSource>>);
+impl AssetKey for SoundtrackKey {
+    type Asset = AudioSource;
 
-impl SoundtrackAssets {
-    pub fn new(asset_server: &AssetServer) -> Self {
-        let mut assets = HashMap::new();
-        assets.insert(
-            SoundtrackAsset::Credits,
-            asset_server.load("audio/soundtracks/Monkeys Spinning Monkeys.ogg"),
-        );
-        assets.insert(
-            SoundtrackAsset::Gameplay,
-            asset_server.load("audio/soundtracks/Fluffing A Duck.ogg"),
-        );
-        Self(assets)
+    fn load(asset_server: &AssetServer) -> AssetMap<Self> {
+        AssetMap(HashMap::from([
+            (
+                SoundtrackKey::Credits,
+                asset_server.load("audio/soundtracks/Monkeys Spinning Monkeys.ogg"),
+            ),
+            (
+                SoundtrackKey::Gameplay,
+                asset_server.load("audio/soundtracks/Fluffing A Duck.ogg"),
+            ),
+        ]))
     }
+}
+
+pub trait AssetKey: Sized {
+    type Asset: Asset;
 
-    pub fn all_loaded(&self, assets: &Assets<AudioSource>) -> bool {
-        self.0.iter().all(|(_, handle)| assets.contains(handle))
+    fn load(asset_server: &AssetServer) -> AssetMap<Self>;
+}
+
+#[derive(Resource, Reflect, Deref, DerefMut)]
+#[reflect(Resource)]
+pub struct AssetMap<K: AssetKey>(HashMap<K, Handle<K::Asset>>);
```

Comment:
I'd implement `From<K, V, const N: usize> for AssetMap`, make `load` return `impl Into<AssetMap>` and then make the `load` implementations return the array directly. Less boilerplate that way.